src/chainstate/stacks/db/blocks.rs

```diff
@@ -1999,16 +2030,19 @@ impl StacksChainState {
     }
 
     /// Delete a microblock's data from the DB
-    fn delete_microblock_data<'a>(
-        tx: &mut DBTx<'a>,
+    fn delete_microblock_data(
+        tx: &mut DBTx,
         microblock_hash: &BlockHeaderHash,
     ) -> Result<(), Error> {
-        // clear out the block data from staging
-        let clear_sql = "DELETE FROM staging_microblocks_data WHERE block_hash = ?1".to_string();
-        let clear_args = [&microblock_hash];
+        let args = [&microblock_hash];
 
-        tx.execute(&clear_sql, &clear_args)
-            .map_err(|e| Error::DBError(db_error::SqliteError(e)))?;
+        // copy into the invalidated_microblocks_data table
+        let copy_sql = "INSERT OR REPLACE INTO invalidated_microblocks_data SELECT * FROM staging_microblocks_data WHERE block_hash = ?1";
+        tx.execute(copy_sql, &args)?;
```

Comment:
No; the point of this PR is to keep that data around indefinitely for the time being.  The problem was that it was getting deleted, and we almost ran into a case where we lost the chain's continuity.

Reply:
I see. If we think that eviction in on our roadmap, should we be preparing the database schema with something like a `created_at` or `block_height`, so we can avoid a migration whenever we think we're ready?

Reply:
I don't think we need to worry too much about eviction.  I'd rather keep all invalid blocks forever just in case (1) we accidentally mark a valid block invalid or (2) their absence somehow causes new nodes from bootstrapping.  To be clear, neither of these should be true, but we've already been bitten by (1) in production.